lib/facter/lvm_support.rb

```diff
@@ -11,34 +13,31 @@
 
 # lvm_vgs: [0-9]+
 #   Number of VGs
-vg_list = []
 Facter.add('lvm_vgs') do
   confine lvm_support: true
 
-  if Facter.value(:lvm_support)
-    vgs = Facter::Core::Execution.execute('vgs -o name --noheadings 2>/dev/null', timeout: 30)
-  end
+  setcode do
+    vgs = Facter::Core::Execution.execute('vgs -o name --noheadings 2>/dev/null', timeout: 30) if Facter.value(:lvm_support)
```

Comment:
I think the condition isn't required because the `lvm_vgs` fact is confined on `lvm_support`
```suggestion
    vgs = Facter::Core::Execution.execute('vgs -o name --noheadings 2>/dev/null', timeout: 30) if Facter.value(:lvm_support)
```

Reply:
Introduced as a bugfix in https://github.com/puppetlabs/puppetlabs-lvm/pull/222/files.
Interestingly this is confined `lvm_support: true`, so shouldn't execute the block in the first place.